Index: misc/tabledrag.js
===================================================================
RCS file: /cvs/drupal/drupal/misc/tabledrag.js,v
retrieving revision 1.38
diff -u -p -r1.38 tabledrag.js
--- misc/tabledrag.js	18 May 2010 06:46:45 -0000	1.38
+++ misc/tabledrag.js	1 Jun 2010 03:37:33 -0000
@@ -594,11 +683,11 @@ Drupal.tableDrag.prototype.updateField =
   var rowSettings = this.rowSettings(group, changedRow);
 
   // Set the row as it's own target.
-  if (rowSettings.relationship == 'self' || rowSettings.relationship == 'group') {
+  if (typeof rowSettings != 'undefined' && (rowSettings.relationship == 'self' || rowSettings.relationship == 'group')) {
     var sourceRow = changedRow;
   }
   // Siblings are easy, check previous and next rows.
-  else if (rowSettings.relationship == 'sibling') {
+  else if (typeof rowSettings != 'undefined' && rowSettings.relationship == 'sibling') {
     var previousRow = $(changedRow).prev('tr').get(0);
     var nextRow = $(changedRow).next('tr').get(0);
     var sourceRow = changedRow;
@@ -625,7 +714,7 @@ Drupal.tableDrag.prototype.updateField =
   }
   // Parents, look up the tree until we find a field not in this group.
   // Go up as many parents as indentations in the changed row.
-  else if (rowSettings.relationship == 'parent') {
+  else if (typeof rowSettings != 'undefined' && rowSettings.relationship == 'parent') {
     var previousRow = $(changedRow).prev('tr');
     while (previousRow.length && $('.indentation', previousRow).length >= this.rowObject.indents) {
       previousRow = previousRow.prev('tr');
@@ -660,8 +749,10 @@ Drupal.tableDrag.prototype.updateField =
     rowSettings.source = rowSettings.target;
   }
 
-  var targetClass = '.' + rowSettings.target;
-  var targetElement = $(targetClass, changedRow).get(0);
+  if (typeof rowSettings != 'undefined') {
+    var targetClass = '.' + rowSettings.target;
+    var targetElement = $(targetClass, changedRow).get(0);
+  }
 
   // Check if a target element exists in this row.
   if (targetElement) {
