diff --git a/media.module b/media.module
index b6caf7e..51167c3 100644
--- a/media.module
+++ b/media.module
@@ -622,21 +622,16 @@ if (!function_exists('file_uri_to_object')) {
    * @return
    *   A file object, or FALSE on error.
    */
-  function file_uri_to_object($uri, $use_existing = FALSE) {
+  function file_uri_to_object($uri, $use_existing = TRUE) {
+    $file = FALSE;
+    $uri = file_stream_wrapper_uri_normalize($uri);
     if ($use_existing) {
-      $query = db_select('file_managed', 'f')
-        ->fields('f', array('fid'))
-        ->condition('uri', $uri)
-        ->execute()
-        ->fetchCol();
-      if (!empty($query)) {
-        $file = file_load(array_shift($query));
-      }
+      // We should always attempt to re-use a file if possible.
+      $files = entity_load('file', FALSE, array('uri' => $uri));
+      $file = !empty($files) ? reset($files) : FALSE;
     }
-    if (!isset($file)) {
+    if (empty($file)) {
       global $user;
-      $uri = file_stream_wrapper_uri_normalize($uri);
-      $wrapper = file_stream_wrapper_get_instance_by_uri($uri);
       $file = new StdClass;
       $file->uid = $user->uid;
       $file->filename = basename($uri);
