diff --git a/olark.js b/olark.js
index b574887..460bf58 100755
--- a/olark.js
+++ b/olark.js
@@ -11,8 +11,10 @@ Drupal.behaviors.olark = function() {
         snippet: Drupal.settings.olark.mail + ' | ' + Drupal.settings.olark.userpage
       });
       api.chat.onReady(function(){
-        //$('#habla_pre_chat_name_input').val(Drupal.settings.olark.name);
-        //$('#habla_pre_chat_email_input').val(Drupal.settings.olark.mail);
+        if ((typeof Drupal.settings.olark.mail != 'undefined') && typeof Drupal.settings.olark.name != 'undefined') {
+          olark('api.visitor.updateEmailAddress', {emailAddress: Drupal.settings.olark.mail});
+          olark('api.visitor.updateFullName', {fullName: Drupal.settings.olark.name});
+        }
       });
     }); 
   }
