diff --git a/og_vocab.module b/og_vocab.module
index fb87590..f597fa2 100644
--- a/og_vocab.module
+++ b/og_vocab.module
@@ -1089,7 +1089,4 @@ function og_vocab_is_group_admin_context() {
   if (strpos($item['path'], 'group/') === 0 && !empty($item['map'][2])) {
     return array('group_type' => $item['map'][1], 'gid' => $item['map'][2]);
   }
-  if (module_exists('og_context') && $context = og_context()) {
-    return $context;
-  }
 }
