diff --git a/og_vocab.module b/og_vocab.module
index 2943c69..f3eefcd 100644
--- a/og_vocab.module
+++ b/og_vocab.module
@@ -959,7 +959,19 @@ function og_vocab_get_accessible_vocabs($entity_type = NULL, $entity = NULL, $ac
     global $user;
     $account = clone $user;
   }
-  if (!$gids = og_get_entity_groups('user', $account)) {
+  
+  if ($setting_vocab_from_user_or_context || true){
+    //Retrieve group from context
+    $context = og_vocab_is_group_admin_context();
+    if($context){
+      $gids[$context['group_type']] = array($context['gid']);
+    }
+  }else{
+    //Retrieve the users groups
+    $gids = og_get_entity_groups('user', $account);
+  }
+  
+  if(!$gids){
     return;
   }
 
