diff --git a/spaces.overrides.inc b/spaces.overrides.inc
index de7af38..aad54c0 100644
--- a/spaces.overrides.inc
+++ b/spaces.overrides.inc
@@ -105,19 +105,22 @@ function spaces_form_alter(&$form, &$form_state, $form_id) {
 
         // For administrators, let them save to either the space or the preset,
         // unless it looks like the buttons may have already been altered.
-        if (isset($form['buttons']) && count(element_children($form['buttons'])) == 2 &&
+        if (isset($form['actions']) && count(element_children($form['actions'])) == 1 &&
             spaces_access_admin_perms(array('administer spaces'))
         ) {
           if ($preset = $space->controllers->variable->get('spaces_preset_' . $space->type)) {
             $preset = spaces_preset_load($preset, $space->type);
-            $form['buttons']['preset'] = array(
+            $form['actions']['preset'] = array(
               '#type' => 'submit',
               '#value' => t('Save to @title', array('@title' => $preset->title)),
             );
-            $form['buttons']['submit']['#value'] = t('Save for @name', array(
+            $form['actions']['reset'] = array(
+              '#type' => 'submit',
+              '#value' => t('Reset to defaults'),
+            );
+            $form['actions']['submit']['#value'] = t('Save for @name', array(
               '@name' => $space->title(),
             ));
-            $form['buttons']['reset']['#weight'] = 100;
           }
         }
       }
@@ -141,8 +144,7 @@ function spaces_system_settings_form_submit($form, &$form_state) {
     $op = isset($form_state['values']['op']) ? $form_state['values']['op'] : '';
 
     // Exclude unnecessary elements.
-    unset($form_state['values']['submit'], $form_state['values']['reset'], $form_state['values']['preset'], $form_state['values']['form_id'], $form_state['values']['op'], $form_state['values']['form_token'], $form_state['values']['form_build_id']);
-
+    form_state_values_clean($form_state);
 
     if ($preset = $space->controllers->variable->get('spaces_preset_' . $space->type)) {
       $preset = spaces_preset_load($preset, $space->type);
