diff --git a/plugins/context_reaction_block.inc b/plugins/context_reaction_block.inc
index fb60982..d56df3d 100644
--- a/plugins/context_reaction_block.inc
+++ b/plugins/context_reaction_block.inc
@@ -565,7 +565,7 @@ class context_reaction_block extends context_reaction {
    * Sort callback.
    */
   static function block_sort($a, $b) {
-    return ($a->weight - $b->weight);
+    return (($a->weight - $b->weight) > 0)? 1 : -1;
   }
 
   /**
