diff --git a/includes/plugins.inc b/includes/plugins.inc
index a6d4bd6..4ac72c3 100644
--- a/includes/plugins.inc
+++ b/includes/plugins.inc
@@ -447,7 +447,7 @@ function ctools_plugin_load_includes($info, $filename = NULL) {
       $all_files[$info['module']][$info['type']] = array();
       // Load all our plugins.
       $directories = ctools_plugin_get_directories($info);
-      $extension = empty($info['info file']) ? $info['extension'] : 'info';
+      $extension = (empty($info['info file'])||($info['extension']!='inc')) ? $info['extension'] : 'info';
 
       foreach ($directories as $module => $path) {
         $all_files[$info['module']][$info['type']][$module] = file_scan_directory($path, '/\.' . $extension . '$/', array('key' => 'name'));
