diff --git a/includes/base.inc b/includes/base.inc
index f804fb2..6d8f33f 100644
--- a/includes/base.inc
+++ b/includes/base.inc
@@ -873,7 +873,12 @@ abstract class MigrationBase {
         }
 
         if ($return == MigrationBase::RESULT_COMPLETED && isset($this->total_successes)) {
-          $overallThroughput = round(60*$this->total_successes / (microtime(TRUE) - $this->starttime));
+          $time = microtime(TRUE) - $this->starttime;
+          if($time > 0){
+            $overallThroughput = round(60*$this->total_successes / $time);
+          }else{
+            $overallThroughput = 9999;
+          }
         }
         else {
           $overallThroughput = 0;
