diff --git a/profiles/openscholar/modules/contrib/biblio/includes/biblio.pages.inc b/profiles/openscholar/modules/contrib/biblio/includes/biblio.pages.inc
index 326a9fb..c326b86 100644
--- a/profiles/openscholar/modules/contrib/biblio/includes/biblio.pages.inc
+++ b/profiles/openscholar/modules/contrib/biblio/includes/biblio.pages.inc
@@ -282,6 +282,8 @@ function biblio_entry($node) {
     );
   }
 
+  drupal_alter('biblio_entry', $entry);
+
   return $entry;
 }
 
