diff --git a/sites/all/modules/contrib/purl/purl.module b/sites/all/modules/contrib/purl/purl.module
index 426bb88..ea11419 100644
--- a/sites/all/modules/contrib/purl/purl.module
+++ b/sites/all/modules/contrib/purl/purl.module
@@ -133,8 +133,8 @@ function purl_init() {
     // TODO - is this still needed?? (Does language_init still modify $_GET['q'])
     // $_GET['q'] = $q = purl_language_strip($_REQUEST['q']);
     $_GET['q'] = $q = $_GET['q'];
-    drupal_path_initialize();
     purl_get_normal_path($q, TRUE);
+    drupal_path_initialize();
   }
 }
 
