diff --git a/modules/contrib/field_redirection/field_redirection.module b/modules/contrib/field_redirection/field_redirection.module
index c97bf7a..0588b2f 100644
--- a/modules/contrib/field_redirection/field_redirection.module
+++ b/modules/contrib/field_redirection/field_redirection.module
@@ -240,6 +240,15 @@ function field_redirection_field_formatter_view($entity_type, $entity, $field, $
       }
     }
     else {
+      // If caching this page, add 'Cache-Control' header before redirecting.
+      $caching_enabled = variable_get('cache', 0);
+      $page_cacheable = drupal_page_is_cacheable();
+      $no_session_cookie = !isset($_COOKIE[session_name()]);
+      if ($caching_enabled && $page_cacheable && $no_session_cookie) {
+        // @see drupal_serve_page_from_cache()
+        drupal_add_http_header('Cache-Control', 'public, max-age=0');
+      }
+
       drupal_goto($path, $options, $response_code);
     }
   }
