diff --git a/modules/contrib/advanced_help/advanced_help.module b/modules/contrib/advanced_help/advanced_help.module
index 1870c35..e894b54 100644
--- a/modules/contrib/advanced_help/advanced_help.module
+++ b/modules/contrib/advanced_help/advanced_help.module
@@ -711,6 +711,12 @@ function _advanced_help_parse_ini() {
   static $cache = NULL;
 
   if (!isset($cache)) {
+    if($topic_info = cache_get('advanced_help_topic_info')){
+      $cache = $topic_info->data;
+    }
+  }
+
+  if (!isset($cache)) {
     $cache = array('topics' => array(), 'settings' => array());
 
     $help_path = drupal_get_path('module', 'advanced_help') . '/modules';
@@ -785,6 +791,7 @@ function _advanced_help_parse_ini() {
       }
     }
     drupal_alter('advanced_help_topic_info', $cache);
+    cache_set('advanced_help_topic_info', $cache);
   }
   return $cache;
 }
@@ -868,7 +875,7 @@ function advanced_help_search_status() {
       }
     }
   }
-  
+
   $last_cron = variable_get('advanced_help_last_cron', array('time' => 0));
   $indexed = 0;
   if ($last_cron['time'] != 0) {
